server/src/main/java/org/elasticsearch/http/AbstractHttpServerTransport.java

```diff
@@ -128,7 +131,19 @@ public HttpInfo info() {
 
     @Override
     public HttpStats stats() {
-        return new HttpStats(httpChannels.size(), totalChannelsAccepted.get());
+        pruneClientStats();
```

Comment:
If I understand this correctly, we only prune stats once the stats are actually requested. Wouldn't this mean that if the stats are never requested then they can grow unbounded?

Reply:
Good point. I'll think about a better way to do that. Probably something based on pruning when new clients connect so that the event that triggers client stats growth is also the one that triggers pruning.

Reply:
@dakrone, in https://github.com/elastic/elasticsearch/pull/64561/commits/ea463660fc127ba470b3ac0ef661ea82b963855b, I added a call to `pruneClientStats` to `serverAcceptedChannel` so that the same event that triggers client stats growth also triggers pruning. That call is throttled to once per minute to better handle situations where large numbers of client are connecting to a cluster.